giskard/utils/xprint.py

```diff
@@ -0,0 +1,115 @@
+from typing import List, Optional
+
+from dataclasses import dataclass
+
+from colorama import Fore
+from colorama import Style as _Style
+
+CHARS_LIMIT = 120
+PLACEHOLDER = "{}"
+
+# Aliasing
+BLACK_COLOR = Fore.LIGHTBLACK_EX
+RED_COLOR = Fore.LIGHTRED_EX
+GREEN_COLOR = Fore.LIGHTGREEN_EX
+YELLOW_COLOR = Fore.LIGHTYELLOW_EX
+BLUE_COLOR = Fore.LIGHTBLUE_EX
+MAGENTA_COLOR = Fore.LIGHTMAGENTA_EX
+CYAN_COLOR = Fore.LIGHTCYAN_EX
+WHITE_COLOR = Fore.LIGHTWHITE_EX
+
+RESET = _Style.RESET_ALL
+BOLD = _Style.BRIGHT
+
+
+@dataclass(frozen=True)
+class Style:
+    color: int = RESET
+    font: int = BOLD
+    design: str = "{reset}{font}{color}{" + PLACEHOLDER + "}{reset}"
+
+
+class Template:
+    def __init__(self, content: str = PLACEHOLDER, pstyles: Optional[List[Style]] = None) -> None:
+        self.content = content
+        self.pstyles = pstyles
+        if not self.pstyles:
+            self.pstyles = [Style()]
+
+    @property
+    def num_placeholders(self):
+        return len(self.content.split(PLACEHOLDER)) - 1
+
+    @property
+    def num_styles(self):
+        return len(self.pstyles)
+
+
+def get_design_templates(template: Template):
+    _design_templates = []
+    for i in range(template.num_placeholders):
+        _design_templates += [
+            template.pstyles[i].design.format(
+                reset=RESET,
+                font=template.pstyles[i].font,
+                color=template.pstyles[i].color,
+                PLACEHOLDER=template.content,
+            )
+        ]
+    return _design_templates
+
+
+def process_template(template: Template):
+    _content_template = template.content.replace(" " + PLACEHOLDER + " ", PLACEHOLDER)
+    for p in [" " + PLACEHOLDER, PLACEHOLDER + " "]:
+        _content_template = _content_template.replace(p, PLACEHOLDER)
+
+    return _content_template.format(*get_design_templates(template)).split(PLACEHOLDER)
+
+
+def style(*args, template: Optional[Template] = None):
+    if not template:
+        template = Template()
+    wraps = process_template(template)
+    out = []
+    for i, _style in enumerate(template.pstyles):
+        if i == template.num_styles - 1 and len(args[i:]) > 1:
```

Comment:
not sure I follow the logic
if there are too many args, we just add the last slice all together ?